src/librbd/api/Image.cc

```diff
@@ -274,6 +274,7 @@ int Image<I>::deep_copy(I *src, librados::IoCtx& dest_md_ctx,
   r = dest->state->open(false);
   if (r < 0) {
     lderr(cct) << "failed to read newly created header" << dendl;
+    delete dest;
```

Comment:
:-1:  -- ```ImageState::open``` will automatically delete the ```ImageCtx``` upon failure [1]

[1] https://github.com/ceph/ceph/blob/master/src/librbd/ImageState.cc#L254